### ISSUE TITLE
BASIC example CIRCLE

### ISSUE THREAD

**Issue by MauriceMEGA65** [MEGA65/mega65-user-guide#207]
PALETTE RESTORE not needed since it's covered by SCREEN CLOSE(?)

**Comment by gurcei** [comment]
I tried to assess this through experimentation. I did the following:
- Entered the program in
- Ran it, here's what the colour palette looked like on that occasion:
![image](https://user-images.githubusercontent.com/20013309/132117995-dabb3a15-b839-4ea2-8fcc-71199c2018ef.png)

- I pressed run-stop, getting the `?BREAK IN 330 READY.` message, so as not to let the program do the SCREEN CLOSE / PALETTE RESTORE steps
- I then manually typed the following:
```
SCREEN CLOSE 1
SCREEN OPEN 1
SCREEN SET 1,1
```
- Upon the graphic screen re-appearing, I note that the colour palette is as the prior program run made it (it hadn't been restored, despite me manually typing in the `SCREEN CLOSE 1`).
- Then I manually typed the `PALETTE RESTORE` command and noticed the colours changed as follows:
![image](https://user-images.githubusercontent.com/20013309/132118071-7257b565-1ec7-4637-8aca-91d3c9aaac62.png)

So from what I've seen, it seems as thoug `SCREEN CLOSE 1` does not automatically perform a `PALETTE RESTORE` for you.

**Comment by MauriceMEGA65** [comment]
What I meant is that I think PALETTE RESTORE at the end of the BASIC program is redundant and can be taken out

**Comment by MauriceMEGA65** [comment]
BitShifter made a comment somewhere that he incorporated PALETTE RESTORE into SCREEN CLOSE.
If that does not seem to work, you found another bug

**Comment by MauriceMEGA65** [comment]
![IMG_0345](https://user-images.githubusercontent.com/75681098/132123048-ebc68c5b-29d6-4532-94d3-0ea094e1cda6.PNG)

**Comment by gurcei** [comment]
Ah ok, good to know.

What seems to be happening is what @Edilbert described to me recently in relation to the "old" screen functionality versus "new" screen functionality.

So it appears that if you were to use new screen functionality for this example, i.e.:

__REPLACE__:
```
140 SCREEN DEF 1,W,H,4
150 SCREEN OPEN 1
160 PALETTE 1,0,0,0,0 : REM BLACK
... all other PALLETE 1,...
230 SCREEN SET 1,1
340 SCREEN CLOSE 1 : BORDER 6 : PALETTE RESTORE
```

__WITH__:
```
140 SCREEN 320,200,4
150 (delete)
160 PALETTE 0,0,0,0,0 : REM BLACK
... all other PALLETE 0,...
230 (delete)
340 SCREEN CLOSE : BORDER 6
```

Then yes, it appears true, the "new" behaviour will do the pallete-restore step for you within `SCREEN CLOSE`.

But the gotchya presently is that this example is written in old-style.

I believe the matter is on @Edilbert 's radar an he is considering update all examples to use the new style.

Part of me also wonders if there's merits in documenting the old-style too, since they both seem to co-exist at present.

Ok, will re-open and await Edilbert's thoughts.

**Comment by MauriceMEGA65** [comment]
Thank you. Will you assign this to BitShifter, or is he already working on it in the background?

**Comment by MauriceMEGA65** [comment]
ps, I think it will be great to have both old and new style incorporated in the manual (can be in different places/examples) because both do work

**Comment by gurcei** [comment]
Ok, will assign him, for whenever he has the appetite to look into it :)

**Comment by MauriceMEGA65** [comment]
ps, I forgot to mention: we already do have both old and new style covered in the SCREEN example.
What now needs to be done is that the remaining graphic examples need to be written in either style instead of a mix.

**Comment by Edilbert** [comment]
I introduced the easy screen setup for following situation:
If only one screen is needed, screen 0 is used.
SCREEN wisth,height depth will then perform following commands (all for screen 0)
- SCREEN DEF
- SCREEN OPEN
- SCREEN SET
- SCREEN CLR
- define default palette

SCREEN CLOSE will then perform:
- SCREEN CLOSE 0
- PALETTE RESTORE

If however SCREEN 1, 2 or 3 is being used, this automatic is not applied to those screens!
Using these screen still requires the old syntax including setting the palette and restoring it.

**Comment by MauriceMEGA65** [comment]
OK, so in the case of CIRCLE, we _do_ need PALETTE RESTORE and my comment can be closed(?)

**Comment by gurcei** [comment]
I reckon, let's wait until the example for CIRCLE gets updated to the new-style and then we can close. @Edilbert, I'm happy to update this CIRCLE example if you like, but if you already had efforts underway with updating the examples, I can leave it in your hands.

**Comment by gurcei** [comment]
Ok, I've updated `CIRCLE` example to use the newer 'simplified' approach to using `SCREEN` command.

I've also updated the docs for the `SCREEN` command to better clarify, emphasise and elaborate on the distinctions between what I've referred to as the "Simplified approach" and the "Detailed approach" to using `SCREEN`.

It's worth a review whenever time permitting, but for now, I'm thinking it's ok to close the ticket.
